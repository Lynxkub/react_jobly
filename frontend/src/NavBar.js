import React  from 'react';
import {NavLink} from 'react-router-dom';

const NavBar = ( { token , username }) => {

  
    
    if(!username){
        return(
            <nav className='navbar  navbar-light bg-dark justify-content-end'>
                 <div className='container-fluid'>
            <NavLink className='navbar-brand fs-1 fw-bold'  exact to='/'>
                Jobly
            </NavLink>
            </div>
        <ul className='nav nav-tabs justify-content-end' id='myTab' role='tablist'>
            
            <li className='nav-item' role='presentation'>
        <NavLink  className='nav-link bg-dark text-light' aria-current='page' exact to='/'>
            Home
        </NavLink>
        </li>
        <li className='nav-item' role='presentation'>
        <NavLink className='nav-link bg-dark text-light' aria-current='page' exact to='/login'>
            Login
        </NavLink>
        </li>
        <li className='nav-item' role='presentation'>
        <NavLink exact className='nav-link bg-dark text-light' aria-current='page' to='/signup'>
            Sign Up 
        </NavLink>
        </li>
        </ul>
        </nav>
        )
    }else {

    
    return (
        <nav className='navbar  navbar-light bg-dark justify-content-end'>
            <div className='container-fluid'>
            <NavLink className='navbar-brand fs-1 fw-bold'  exact to='/'>
                Jobly
            </NavLink>
            </div>
    <ul className='nav nav-tabs justify-content-end ' id='myTab' role='tablist'>
        <li className='nav-items' role='presentation'>
        <NavLink className='nav-link bg-dark text-light' aria-current='page' exact to='/'>
            Home
        </NavLink>
        </li>
        <li className='nav-items' role='presentation'>
        <NavLink className='nav-link bg-dark text-light' aria-current='page'  exact to='/companies'>
            Companies
        </NavLink>
        </li>
        <li className='nav-items' role='presentation'>
        <NavLink className='nav-link bg-dark text-light' aria-current='page'  exact to='/jobs'>
            Jobs
        </NavLink>
        </li>
        <li className='nav-items' role='presentation'>
        <NavLink className='nav-link bg-dark text-light' aria-current='page'   exact to='/profile'>
            Profile
        </NavLink>
        </li>
        <li className='nav-items' role='presentation'>
        <NavLink className='nav-link bg-dark text-light' aria-current='page' exact to='/logout'>
            {username} Log Out
        </NavLink>
        </li>
        </ul>
        </nav>
    )
    }
}


export default NavBar;