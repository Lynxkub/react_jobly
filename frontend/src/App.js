import React , {useEffect, useState} from 'react';
import './App.css';
import NavBar from './NavBar';
import Routes from './Routes';
import JoblyApi from './api';



function App() {
 
  const [userToken , setUserToken] = useState(localStorage.getItem('token'))
  const [username , setUsername] = useState(localStorage.getItem('username'))

  useEffect(() => {
    localStorage.setItem('token' , userToken)
  } , [userToken])


  JoblyApi.token = localStorage.getItem('token');
  
  const jobsAppliedTo = localStorage.getItem('jobs');
  if(!jobsAppliedTo){
    localStorage.setItem('jobs' , JSON.stringify([' ']));
  }
 

  const signUp = async (newUser) => {
  setUserToken(await JoblyApi.register(newUser));
  localStorage.setItem('token' , userToken)
  setUsername(localStorage.getItem('username'));
}

  const login = async (user) => {
    setUserToken(await JoblyApi.login(user))
    localStorage.setItem('token' , userToken)
    setUsername(localStorage.getItem('username'));
  }

  const logout =  () => {
    setUserToken(undefined);
    setUsername(localStorage.getItem('username'));
  

  }



  return (
    <div className='fill-window'>
    <NavBar token = {userToken} username = {username} />
    <Routes signUp = {signUp} login = {login} logout = {logout} token = {userToken} />
    </div>
  );
}

export default App;

