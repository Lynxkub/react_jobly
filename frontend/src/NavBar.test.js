import React from 'react';
import {render , fireEvent} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import NavBar from './NavBar';
import App from './App';
import { sign } from 'jsonwebtoken';


test('it renders without crashing' , () => {
    render(
    <MemoryRouter>
    <NavBar />
    </MemoryRouter>
    )
})

it('matches snapshot' , () => {
    const {asFragment} =  render(
        <MemoryRouter>
        <NavBar />
        </MemoryRouter>
        )
    expect(asFragment()).toMatchSnapshot();
    
})

it('follows link to companies page correctly after logging in correctly' , () => {
    const {getByText , getByTestId} =  render(
        <MemoryRouter initialEntries={['/']}>
        <App />
        </MemoryRouter>
        )

        // login to website before following companies link
    const login = getByText('Login');
    fireEvent.click(login);
    
  
    const username = getByTestId('username');
    const password = getByTestId('password');
    const signInButton = getByText('Submit')
    fireEvent.change(username , {target : {value : 'test'}});
    fireEvent.change(password , {target : {value : 'testtest'}});
    fireEvent.click(signInButton);


    // wait for api call to be made and rendered after the link is clicked

    setTimeout(() => {
        const companies = getByText('Companies');
        fireEvent.click(companies);
        const firstCompany = getByText('Anderson , Arias and Morrow');
        expect(firstCompany).toBeInTheDocument();
    } , 1000)
})