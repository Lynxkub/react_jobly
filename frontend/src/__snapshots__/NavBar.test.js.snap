// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Home
    </a>
    <a
      href="/login"
    >
      Login
    </a>
    <a
      href="/signup"
    >
      Sign Up
    </a>
  </div>
</DocumentFragment>
`;
